pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -927,9 +929,44 @@ class MCPServerConfig(BaseModel):
     ]
 
 
+def _expand_env_vars(value: Any) -> Any:
+    """Recursively expand environment variables in a JSON structure.
+
+    Environment variables can be referenced using ${VAR_NAME} syntax.
+
+    Args:
+        value: The value to expand (can be str, dict, list, or other JSON types).
+
+    Returns:
+        The value with all environment variables expanded.
+
+    Raises:
+        ValueError: If an environment variable is not defined.
+    """
+    if isinstance(value, str):
+        # Find all environment variable references in the string
+        env_var_pattern = re.compile(r'\$\{([^}]+)\}')
+        matches = env_var_pattern.findall(value)
+
+        for var_name in matches:
+            if var_name not in os.environ:
+                raise ValueError(f'Environment variable ${{{var_name}}} is not defined')
+            value = value.replace(f'${{{var_name}}}', os.environ[var_name])
```

Comment:
Insetad of `findall` + `replace`, let's use `re.sub` with a function that directly returns the new value